### PR TITLE
Add first python binding test

### PR DESCRIPTION
**Pull request by andrew-catalyst**
Run python tests on CI. Currently just on linux build

### CODE REVIEW

File: omf-python/tests/test_reader.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        omf_file = path.join(self.examples_dir, "pyramid/pyramid.omf")
+
+        # When
+        reader = omf_python.Reader(omf_file)
+
+        # Then
+        self.assertEqual(reader.project.name, "Pyramid")
+        
+    def test_should_raise_expected_file_not_found_exception(self) -> None:
+        # TODO: currently raises pyo3_runtime.PanicException, should raise a sensible exception here
```
Comment (andrew-catalyst) [inline omf-python/tests/test_reader.py]: Noticed this, we should flesh this out when its fixed.

**Comment by antbrown** [comment]
Can ignore `CI / Check (push)` failure in this PR, I've fixed it in PR4.

Is there a quick way I can run the test locally?

**Comment by andrew-catalyst** [comment]
> Can ignore `CI / Check (push)` failure in this PR, I've fixed it in PR4.
> 
> Is there a quick way I can run the test locally?

Yes just run `pytest` in the omf-python/ directory ( I added it to requirements.txt )

**Comment by antbrown** [comment]
I'll have a go at fixing up the 

```
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }
``` 

so we can have the `test_should_raise_expected_file_not_found_exception`

**Comment by antbrown** [comment]
@andrew-catalyst are you able to check the changes I made to implement the file not found test? feel free to change anything I've put in there

**Comment by antbrown** [comment]
I didn't know whether to do the test using the `with ...` method, or a `try catch` as this person suggests: https://stackoverflow.com/a/23947350

**Comment by andrew-catalyst** [comment]
> I didn't know whether to do the test using the `with ...` method, or a `try catch` as this person suggests: https://stackoverflow.com/a/23947350

That's fine. I usually use `with self.assertRaises(ExceptionType): ...`.

**Comment by andrew-catalyst** [comment]
I tweaked the exception test, if that's ok maybe we can merge this

**Comment by antbrown** [comment]
> I tweaked the exception test, if that's ok maybe we can merge this

Absolutely, thanks for showing me the better way :+1:

**Review by antbrown — APPROVED** [review]

**Pull request merged**